cmd/leanix-k8s-connector/main.go

```diff
@@ -120,126 +115,139 @@ func KubernetesScan(debugLogBuffer *bytes.Buffer) (response *leanix.SelfStartRes
 			return nil, err
 		}
 	}
-
 	log.Debugf("Kubernetes master from config: %s", config.Host)
+	kubernetesObjects := make([]mapper.KubernetesObject, 0)
+	if startResponse.BindingKey.ConnectorId == "leanix-k8s-v3-connector" {
+		log.Info("Using new k8s-v3 mapper")
+		mappedObject, err := newmapper.ScanWorker(startResponse.ConnectorConfiguration.ClusterName, config)
+		if err != nil {
+			return nil, err
+		}
+		kubernetesObjects = append(kubernetesObjects, mappedObject...)
 
-	kubernetesAPI, err := kubernetes.NewAPI(config)
-	if err != nil {
-		return nil, err
-	}
-	dynClient, err := dynamic.NewForConfig(config)
-	if err != nil {
-		log.Error(err)
-	}
-
-	log.Debug("Get blacklist namespaces list...")
-	blacklistedNamespacesList := viper.GetStringSlice(blacklistNamespacesFlag)
-	blacklistedNamespaces, err := kubernetesAPI.Namespaces(blacklistedNamespacesList)
-	if err != nil {
-		return nil, err
-	}
-	log.Debug("Getting blacklist namespaces list done.")
-	log.Infof("Namespace blacklist: %v", reflect.ValueOf(blacklistedNamespaces).MapKeys())
+		_, err = leanix.UpdateInProgressStatus(startResponse.ProgressCallbackUrl, "Mapping of kubernetes objects with new version done. Preparing ldif.")
+		if err != nil {
+			log.Infof("New Kubernetes Scan: After Mapping: Failed to update progress[%s] to Integration Hub", leanix.IN_PROGRESS)
+		}
+	} else {
```

Comment:
We could also think about moving the code for the old pattern into a separate function so it's more readable

Reply:
Good idea, just did that, definitely makes the main file a bit more readable.